src/lib/operators/abstract_operator.hpp

```diff
@@ -116,6 +119,11 @@ class AbstractOperator : public std::enable_shared_from_this<AbstractOperator>,
   std::shared_ptr<const Table> left_input_table() const;
   std::shared_ptr<const Table> right_input_table() const;
 
+  // Consumer tracking
```

Comment:
We should doc that `clear_output` is only called if a consumer has been registered, and that, as a result, it is never called on the top (result) operator.

Reply:
Shall we make `clear_output()` protected or remove it altogether?

Reply:
Let's discuss this later once we have a clear picture of the rest.